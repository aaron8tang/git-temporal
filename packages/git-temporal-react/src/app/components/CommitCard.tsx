import React from 'react';
import { style } from 'app/styles';

import { ICommit } from 'app/interfaces';

import { EpochDate } from 'app/components/EpochDate';
import { AddedDeleted } from 'app/components/AddedDeleted';
import { CommitBody } from 'app/components/CommitBody';

export interface CommitCardProps {
  commit: ICommit;
  index?: number;
  isExpanded?: boolean;
  onClick?: (evt, commit: ICommit, index: number) => void;
  style?: string | object;
}

const defaultCardStyle = {
  _extends: ['card', 'block'],
  borderBottom: '1px solid whitesmoke',
};

const dateLineStyle = {
  _extends: ['smallerText', 'block'],
};
const messageStyle = {
  _extends: ['largerText', 'block'],
  marginLeft: 5,
};
const authorStyle = {
  _extends: ['normalText', 'block'],
  marginLeft: 5,
};

const constrainedBodyStyle = {
  maxHeight: 70,
  overflow: 'hidden',
  marginBottom: 10,
  boxShadow: 'rgba(245, 245, 245, 0.5) 0px -34px 11px -18px inset',
};

export const CommitCard = (props: CommitCardProps): JSX.Element => {
  const { commit, isExpanded } = props;
  const outerOverrideStyle = isExpanded ? 'selected' : {};
  const bodyStyle =
    isExpanded || commit.body.trim() === '' ? {} : constrainedBodyStyle;
  return (
    <div
      style={style(defaultCardStyle, props.style, outerOverrideStyle)}
      onClick={evt => props.onClick(evt, commit, props.index)}
    >
      <div style={style(dateLineStyle)}>
        <EpochDate epochTime={commit.authorDate} />
        <span style={{ marginLeft: 60 }}>{commit.hash}</span>
        <AddedDeleted
          linesAdded={commit.linesAdded}
          linesDeleted={commit.linesDeleted}
        />
      </div>
      <div style={style(messageStyle)}>{commit.message}</div>
      <div style={style(bodyStyle)}>
        <CommitBody text={commit.body} />
      </div>
      <div style={style(authorStyle)}>
        Authored by {commit.authorName} {commit.relativeDate}
      </div>
    </div>
  );
};
