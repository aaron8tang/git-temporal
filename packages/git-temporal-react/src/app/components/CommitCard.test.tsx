import React from 'react';
import { shallow } from 'enzyme';

import { CommitCard } from './CommitCard';

const testProps = {
  commit: {
    id: '408e8ce22ce7436e88a9977b4dc99fc3c431fc67',
    authorName: 'MichaÃ«l Zasso',
    relativeDate: '3 days ago',
    authorDate: 1537212012,
    message: 'doc: update maintaining V8 guide',
    body:
      'Replace references to the outdated `update-v8` tool to its replacementin `node-core-utils`PR-URL: https://github.com/nodejs/node/pull/22913Reviewed-By: Richard Lau <riclau@uk.ibm.com>Reviewed-By: Colin Ihrig <cjihrig@gmail.com>Reviewed-By: Weijia Wang <starkwang@126.com>Reviewed-By: Sakthipriyan Vairamani <thechargingvolcano@gmail.com>Reviewed-By: Joyee Cheung <joyeec9h3@gmail.com>Reviewed-By: Luigi Pinca <luigipinca@gmail.com>Reviewed-By: James M Snell <jasnell@gmail.com>',
    hash: '408e8ce22c',
    linesAdded: 8,
    linesDeleted: 8,
    files: [
      {
        name: 'doc/guides/maintaining-V8.md',
        linesAdded: 8,
        linesDeleted: 8,
      },
    ],
  },
};

describe('components/CommitCard', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<CommitCard {...testProps} />);
    });

    test('it should match snapshot (bars should be at 50%)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
