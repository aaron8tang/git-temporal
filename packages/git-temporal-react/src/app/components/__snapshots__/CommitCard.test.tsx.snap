// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CommitCard when rendered with required props it should match snapshot (bars should be at 50%) 1`] = `
<div
  style={
    Object {
      "background": "#FEFEFE",
      "borderRadius": 7,
      "color": "#333333",
      "display": "block",
      "marginBottom": "7px",
      "marginRight": 0,
      "padding": 0,
    }
  }
>
  <Selectable
    onClick={[Function]}
    onDoubleClick={[Function]}
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 11,
          "fontWeight": "normal",
          "lineHeight": 1.35,
          "marginBottom": "7px",
        }
      }
    >
      <EpochDateTime
        displayOptions={
          Object {
            "minute": "numeric",
            "month": "long",
          }
        }
        value={1537212012}
      />
      <span
        style={
          Object {
            "flexGrow": 1,
            "textAlign": "center",
          }
        }
      >
        408e8ce22c
      </span>
      <AddedDeleted
        linesAdded={8}
        linesDeleted={8}
      />
    </div>
    <div
      style={
        Object {
          "display": "block",
          "fontSize": 14,
          "fontWeight": "normal",
          "lineHeight": 1.35,
          "marginBottom": "7px",
          "marginLeft": 5,
        }
      }
    >
      doc: update maintaining V8 guide
    </div>
    <CommitBody
      text="Replace references to the outdated \`update-v8\` tool to its replacementin \`node-core-utils\`PR-URL: https://github.com/nodejs/node/pull/22913Reviewed-By: Richard Lau <riclau@uk.ibm.com>Reviewed-By: Colin Ihrig <cjihrig@gmail.com>Reviewed-By: Weijia Wang <starkwang@126.com>Reviewed-By: Sakthipriyan Vairamani <thechargingvolcano@gmail.com>Reviewed-By: Joyee Cheung <joyeec9h3@gmail.com>Reviewed-By: Luigi Pinca <luigipinca@gmail.com>Reviewed-By: James M Snell <jasnell@gmail.com>"
    />
    <CommitCardFiles
      files={
        Array [
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "doc/guides/maintaining-V8.md",
          },
        ]
      }
      style={
        Object {
          "_extends": Array [
            "block",
            "normalText",
          ],
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 20,
        }
      }
    />
    <div
      style={
        Object {
          "display": "block",
          "fontSize": 12,
          "fontWeight": "normal",
          "lineHeight": 1.35,
          "marginBottom": "7px",
          "marginLeft": 5,
        }
      }
    >
      Authored by 
      MichaÃ«l Zasso
       
      3 days ago
    </div>
  </Selectable>
</div>
`;
